Add external user id auth support

Jeasmine opened the pull request:
* Add new methods for external id auth

Jeasmine committed:
Add external user id auth support

* Add new methods for external id auth

rgomezp commented on index.js:
```diff
@@ -444,6 +444,15 @@ export default class OneSignal {
         RNOneSignal.setExternalUserId(externalId, callback);
     }
 
+    static setExternalUserId(externalId, authHashToken, callback) {
```
```js
    static setExternalUserId(externalId, authHashToken, callback=function(){}) {
```

instead of 

```js
if (callback === undefined)
            callback = function(){};
```

rgomezp commented on index.js:
```diff
@@ -435,15 +435,18 @@ export default class OneSignal {
         RNOneSignal.setExternalUserId(externalId, callback);
     }
 
+    static setExternalUserId(externalId, authHashToken, callback=function(){}) {
```
Function overloading doesn't work like this in JS

jkasten2 replied:
Function defaulting is an ES6 feature however I believe we still need to support ES5 for now.
We should keep this code to default.
```javascript
if (callback === undefined)
   callback = function(){};
```

jkasten2 replied:
We need to use `typeof` to see what the 2nd param really is so we don't break anyone's existing usage of `setExternalUserId`.

rgomezp replied:
updated

rgomezp committed:
setExternalUserId Implementation

Motivation: we want to provide folks the ability to pass the auth hash to `setExternalUserId` while maintaining backwards compatibility. Since we can't overload functions, we implement dynamic variables here by checking what the type of the second argument is. If it is a function, we know it is the callback and no auth hash was provided. If not, it must be a string meaning the callback was omitted and we default to an empty function OR we pass it in as the third argument.

rgomezp committed:
Nit: update app id to RN Demo App

rgomezp commented on index.js:
```diff
@@ -435,13 +435,16 @@ export default class OneSignal {
+            return;
+        }
+
+        RNOneSignal.setExternalUserId(externalId, varArg1, varArg2 || function(){});
```
I imagine this exists on ES5 but not 100 sure

rgomezp replied:
I think short circuit evaluation is a JS standard

jkasten2 replied:
yes, this works in ES5

emawby reviewed (approved):
lgtm!
